### ISSUE TITLE
[MulticoreJIT] Single file deployment does not support ProfileOptimization

### ISSUE THREAD

**Issue by FixBo** [dotnet/runtime#108270]
Create app with profile enable

ProfileOptimization.SetProfileRoot(AppDomain.CurrentDomain.BaseDirectory);
ProfileOptimization.StartProfile("profile");

Publish as selfcontained single file

dotnet publish --sc -p:PublishSingleFile=true

Run application

The "profile" is created with a size of only 64 bytes

**Comment by agocke** [comment]
> The "profile" is created with a size of only 64 bytes
> According to my research, it's because of this check

Yup, looks likely. At the moment I would say that it's not supported w/ single-file. I can't promise we'll get to this any time soon.
